src/games/strategy/triplea/ui/RouteOptimizer.java

```diff
@@ -145,70 +138,26 @@ public Point getLastEndPoint() {
   }
 
   /**
-   * Gives a List of Point arrays (Routes) which are the offset equivalent of the given points
-   * Size may vary depending on MapProperties
+   * Matrix Transpose method to transpose the 2dimensional point list
    * 
    * @param points A Point array
-   * @return Offset Point Arrays
+   * @return Offset Point Arrays including points
    */
-  private List<Point[]> getAlternativePoints(Point... points) {
-    List<Point[]> alternativePoints = new ArrayList<>();
-    if (isInfiniteX || isInfiniteY) {
-      int altArrayCount = getAlternativePointArrayCount();
-      for (int i = 0; i < altArrayCount; i++) {
-        alternativePoints.add(new Point[points.length]);
-      }
-      int counter = 0;
-      for (Point point : points) {
-        if (isInfiniteX) {
-          alternativePoints.get(0)[counter] = new Point(point.x - mapWidth, point.y);
-          alternativePoints.get(1)[counter] = new Point(point.x + mapWidth, point.y);
-        }
-        if (isInfiniteY) {
-          int index = altArrayCount == maxAdditionalScreens ? 2 : 0;
-          alternativePoints.get(index)[counter] = new Point(point.x, point.y - mapHeight);
-          alternativePoints.get(index + 1)[counter] = new Point(point.x, point.y + mapHeight);
-        }
-        if (isInfiniteX && isInfiniteY) {
-          alternativePoints.get(4)[counter] = new Point(point.x - mapWidth, point.y - mapHeight);
-          alternativePoints.get(5)[counter] = new Point(point.x - mapWidth, point.y + mapHeight);
-          alternativePoints.get(6)[counter] = new Point(point.x + mapWidth, point.y - mapHeight);
-          alternativePoints.get(7)[counter] = new Point(point.x + mapWidth, point.y + mapHeight);
+  public List<Point[]> getAllPoints(Point... points) {
```

Comment:
No test case for this one? Is it an omission, something that would be particularly difficult to test?

If the latter, it can be made pretty easy to test by updating the API slightly:
```
public List<Point[]> getAllPoints(Point... points) {
  this.getAllPoints(point -> getpossiblePoints(point), points);
}

@VisibleForTesting
List<Point[]> getAllPoints(Function<Point,List<Point2D>> possiblePointsFunction, Point... points) {
 List<Point[]> allPoints = new ArrayList<>();
 +    for (int i = 0; i < points.length; i++) {
 +      List<Point2D> subPoints = possiblePointFunction.apply(points[i]);
 +      for (int y = 0; y < subPoints.size(); y++) {
 +        if (i == 0) {
 +          allPoints.add(new Point[points.length]);
          }
 +        allPoints.get(y)[i] = getPoint((subPoints.get(y)));
      }
```

With the above you can then send in a mock or a stubbed function for the possible point function. Just one thought on how we can increase the test coverage... I'm curious to know your thoughts on how and if we can/should test the `getAllPoints` method.